ts/src/pro/xt.ts

```diff
@@ -0,0 +1,1062 @@
+//  ---------------------------------------------------------------------------
+
+import xtRest from '../xt.js';
+import { ArrayCache, ArrayCacheBySymbolById, ArrayCacheByTimestamp } from '../base/ws/Cache.js';
+import { Precise } from '../base/Precise.js';
+import { Int } from '../base/types.js';
+import Client from '../base/ws/Client.js';
+
+//  ---------------------------------------------------------------------------
+
+export default class xt extends xtRest {
```

Comment:
First we're missing `handleErrorMessage` second I think we should try to handle this logic (upon getting the error request a new key)
![image](https://github.com/ccxt/ccxt/assets/43336371/ee36d820-2cb1-4691-a533-9cd900c2bac3)

Reply:
Step 2 is done right here https://github.com/ccxt/ccxt/pull/17814/files#r1204897903

Reply:
I added the method, there's nothing like this specified under the `spot` documentation, so I set `isContract` to always be `true`, another option would be to just wait for the next call to `getListenKey` to get called organically before getting a new listen key